staging/src/k8s.io/apimachinery/pkg/util/sets/set.go

```diff
@@ -0,0 +1,212 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sets
+
+import (
+	"sort"
+)
+
+// Set is a set of the same type elements, implemented via map[comparable]struct{} for minimal memory consumption.
+type Set[T comparable] map[T]Empty
```

Comment:
nit: Using this as `sets.Set[string]` stutters. How about `sets.Generic[string]`? That's what I picked as name (see https://github.com/kubernetes/kubernetes/pull/112852).

Reply:
I'll leave it up to @weilaaa, I can't decide, either name works for me.

Reply:
`sets.Generic` is little wired when used like `func (s Generic[T]) Insert(items ...T) Generic[T] {}`. This makes it confused for readers to understand.